Fixing selector

rhys-childs opened the pull request:
Fixes #76 
Loops through each matching element rather than the first in the document

rhys-childs committed:
fix multiple matching element bug

fixes query selector with miltiple matching elements by selecting all of the duplicates and processing each

rhys-childs committed:
resolving duplicate click event listeners

internettrans commented on src/use-text-as-image.hook.js:
```diff
@@ -16,8 +16,10 @@ export function useTextAsImage({processSerializedElement = noop, fontFamily = nu
       performCommandWithValue(url)
-      const imgElement = document.querySelector(`img[src="${url}"]`)
-      processImgElement(imgElement, text)
+      const imgElement = document.querySelectorAll(`img[src="${url}"]`)
```
Instead of selecting all the images and then skipping the ones that have already been processed, I think we could select only the one that hasn't been processed.

```suggestion
      const imgElement = document.querySelector(`img[src="${url}"]:not([data-text-as-image])`)
```

Using the `data-text-as-image` attribute would also prevent us from having to create the extra `data-has-listener` attribute, too.

rhys-childs replied:
That is a much more efficient solution, thank you!

internettrans reviewed (commented):
Thanks for the good bug description and pull request to fix the problem. I think there's an easier way to fix this bug - see my comment below.

rhys-childs committed:
updating query selector

using a more refined query Selector rather than looping through all elements found via querySelectorAll

rhys-childs committed:
reverting changes

internettrans reviewed (approved):
lgtm

internettrans commented:
Released in https://github.com/CanopyTax/bandicoot/releases/tag/v3.0.3
